### ISSUE TITLE
Create practical examples.

### ISSUE THREAD

**Issue by desmondmorris** [desmondmorris/node-twitter#63]
- [x] Streams!
- [x] Proxies
- [x] Tweet
- [x] Search
- [x] Media

**Comment by rcc26** [comment]
actually I'm having a bit of a problem understading how to stream posts from a certain user. I tried

    var client = new twitterStream({
		consumer_key: 'xxx',
		consumer_secret: 'xxx',
		access_token_key: 'xxx',
		access_token_secret: 'xxx'
	});
	
       var user = "some_screen_name";

	client.stream('statuses/user_timeline?screen_name='+user, {track: 'javascript'}, function(stream) {

		stream.on('data', function(tweet) {
			console.log("inside stream:",tweet);
		});

	});

but the service failed to run despite any attempts to change the user variable. Is this the correct way to procede Desmond?

**Comment by desmondmorris** [comment]
@rcc26 I do not think the endpoint to passed exists.  I believe you are looking for `user` ([Documentation](https://dev.twitter.com/streaming/reference/get/user)).  Does not look like it accepts a `screen_name` parameter.

```javascript

var Twitter = require('twitter');
var client = new Twitter({
    consumer_key: 'xxx',
    consumer_secret: 'xxx',
    access_token_key: 'xxx',
    access_token_secret: 'xxx'
});

var user = "some_screen_name";

client.stream('user', function(stream) {
    stream.on('data', function(tweet) {
        console.log("inside stream:",tweet);
    });

});
```

**Comment by rcc26** [comment]
@desmondmorris so this only allows self-tweeted (by user access token) posts to be streamed? Is there any way to create streams for other users at all?

**Comment by desmondmorris** [comment]
@rcc use the "status/filter" endpoint with the follow parameter.  Notice the follow parameter takes userid's not screennames. 

```javascript
var user_id = "123456";

client.stream('statuses/filter', {follow: user_id}, function(stream) {
  stream.on('data', function(tweet) {
    console.log("inside stream:",tweet);
  });

  stream.on('error', function(error) {
    console.log(error);
  });
});
````

**Comment by rcc26** [comment]
and why no support for screen names? :'(

**Comment by desmondmorris** [comment]
@rcc26 - its a limitation of the API -https://dev.twitter.com/streaming/overview/request-parameters#follow .  You can get the "users/show" endpoint to get the id - https://dev.twitter.com/rest/reference/get/users/show.

**Comment by rcc26** [comment]
@desmondmorris Tyvm desmond. Does this keep track of repeated stream instances or do I have to handle it myself? If I try and create a stream for a {follow:user_id} that already exists is the initial stream replaced or does it get duplicated?

**Comment by Limero** [comment]
Any progress on authentication? Or isn't that feature implemented yet? I see open pull request #85

**Comment by desmondmorris** [comment]
@Limero authentication (user and application) are both available in the library.

(3-legged auth)[https://dev.twitter.com/oauth/3-legged] is out of the scope of this library at the moment.
